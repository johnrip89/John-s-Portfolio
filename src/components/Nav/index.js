import React from 'react';
import { capitalizeFirstLetter } from '../../utils/helpers';

function Nav() {   

    return (
        <div class="col-7">
            <ul class="nav nav-tabs">
                <li class="nav-item">
                    <a class="nav-link" href="">Home</a>
                </li>
                <li class="nav-item">
                    <a class="nav-link" href="#contact">Contact</a>
                </li>
                <li class="nav-item">
                    <a class="nav-link" href="#projects">Projects</a>
                </li>
            </ul>
        </div>
    );
}

export default Nav;