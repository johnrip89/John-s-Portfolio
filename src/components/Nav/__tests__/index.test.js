import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

const mockProjectsSelected = jest.fn();
const mockSetProjectsSelected = jest.fn();
const mockContactSelected = jest.fn();
const mockSetContactSelected = jest.fn();
const mockResumeSelected = jest.fn();
const mockSetResumeSelected = jest.fn();

afterEach(cleanup);

describe('Nav component', () => {
    it('renders', () => {
        render(<Nav
            setContactSelected={mockSetContactSelected}
            projectsSelected={mockProjectsSelected}
            setProjectsSelected={mockSetProjectsSelected}
            resumeSelected={mockResumeSelected}
            setResumeSelected={mockSetResumeSelected}
        />);
    })

    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<Nav
            contactSelected={mockContactSelected}
            setContactSelected={mockSetContactSelected}
            projectsSelected={mockProjectsSelected}
            setProjectsSelected={mockSetProjectsSelected}
            resumeSelected={mockResumeSelected}
            setResumeSelected={mockSetResumeSelected}
        />);

        expect(asFragment()).toMatchSnapshot();
    });
})

describe('onClick events', () => {
    it('calls the click handler when clicked', () => {
        const { getByText } = render(<Nav
            contactSelected={mockContactSelected}
            setContactSelected={mockSetContactSelected}
            projectsSelected={mockProjectsSelected}
            setProjectsSelected={mockSetProjectsSelected}
            resumeSelected={mockResumeSelected}
            setResumeSelected={mockSetResumeSelected}
        />);
        fireEvent.click(getByText('About'))
        fireEvent.click(getByText('Contact'))
        fireEvent.click(getByText('Resume'))
        fireEvent.click(getByText('Projects'))

        expect(mockSetContactSelected).toHaveBeenCalledTimes(3);
    });
})