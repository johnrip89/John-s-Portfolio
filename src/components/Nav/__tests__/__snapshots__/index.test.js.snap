// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="col-7"
  >
    <ul
      class="nav nav-tabs"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href=""
        >
          About Me
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="#contact"
        >
          Contact
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="#projects"
        >
          Portfolio
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href=""
        >
          Resume
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
